### ISSUE TITLE
"unconfigured table" error returned when query data beyond table-manager.retention-period

### ISSUE THREAD

**Issue by Nevermore328** [cortexproject/cortex#3143]
Hi, here's my situation:
My cortex table-manager.retention-period=2w,  when I call cortex-querier's interface "/api/prom/api/v1/query_range" to query data less than 2 weeks，it returns the correct data, but when I query data beyond 2 weeks, it returns "unconfigured table" error just like this:
{"status":"error","errorType":"internal","error":"expanding series: unconfigured table cortex_2639"}
So I'm confused why cortex does not return data in this case, I mean I query 1 month data but but cortex just has 2 weeks data why not just return 2 weeks data?Did I miss some configs or something else?Please tell me, Thanks very much.

**Comment by pstibrany** [comment]
Hi, thank you for your report. What does your schema file look like? Is it possible that `from:` field has date earlier that 2 weeks ago (before you have any data)? That would explain why Cortex is trying to access tables that it expects should exist, but don't.

**Comment by Nevermore328** [comment]
Yes,It looks like this:
configs:
  - from: 2000-01-01
    store: cassandra
    object_store: cassandra
    schema: v10
    index:
      prefix: cortex_
      period: 168h
    chunks:
      prefix: cortex_chunks_
      period: 168h
    row_shards: 16

So, this from: field tells cortex the beginning time of the data, I see, Thanks very much. By the way, is there any doc about the  schema file, because I didn't find one.

**Comment by Nevermore328** [comment]
I've changed the `from:` value to 2020-09-09, but still got the "unconfigured table" error, did I miss something?

**Comment by pstibrany** [comment]
Hi, have you restarted all Cortex components (especially queriers) after changing schema? I cannot reproduce the behaviour you're seeing -- when I try to query data before `from:`, I simply get no results, but also no errors.

I also cannot find `unconfigured table` error message in our codebase. However `cortex_2639` would correspond to time range `30 July 2020` to week later. Cortex should recognize that it has no data from that time based on the schema config.

**Comment by Nevermore328** [comment]
Thanks for your reply.
I think `unconfigured table` error must be returned by Cassandra, so you cannot find it in Cortex codebase. 
And after changing schema yesterday, I only restarted Cortex-table-manager, it still returned  `unconfigured table` error. But just now, I restarted other Cortex components, it returned the error below:
`{"status":"error","errorType":"internal","error":"expanding series: no chunk table found for time 1599594394.426"}`
I found the `no chunk table found for time` error in Cortex source code schema_config.go like this:
```
func (cfg SchemaConfig) ChunkTableFor(t model.Time) (string, error) {
	for i := range cfg.Configs {
		if t >= cfg.Configs[i].From.Time && (i+1 == len(cfg.Configs) || t < cfg.Configs[i+1].From.Time) {
			return cfg.Configs[i].ChunkTables.TableFor(t), nil
		}
	}
	return "", fmt.Errorf("no chunk table found for time %v", t)
} 
```
It seems that if the query start time is earlier than From.Time,  it throws this error?
